src/manifest.c

```diff
@@ -1395,3 +1492,73 @@ struct file *search_file_in_manifest(struct manifest *manifest, const char *file
 	}
 	return NULL;
 }
+
+/* Ideally have the manifest sorted already by this point */
+struct file **manifest_files_to_array(struct manifest *manifest)
+{
+	struct file **array;
+	struct list *iter;
+	struct file *file;
+	int i = 0;
+	int numfiles = manifest->filecount;
+
+	if (!manifest) {
+		fprintf(stderr, "Manifest is NULL!\n");
+		return NULL;
+	}
+
+	array = malloc(sizeof(struct file *) * numfiles);
+
+	iter = list_head(manifest->files);
+	while (iter) {
+		file = iter->data;
+		iter = iter->next;
+		array[i] = file;
+		i++;
+	}
+	return array;
+}
+
+void print_manifest_array(struct file **array, int filecount)
+{
+	for (int i = 0; i < filecount; i++) {
+		printf("[%d]/%d %s\n", i, filecount, array[i]->filename);
+	}
+	printf("SIZE: %d\n", filecount);
```

Comment:
I think it would make more sense to print something like `NUMBER OF FILES: filecount` since this isn't the actual size of all the files in the manifest, just the number of them.